src/main/webapp/app/shared/modal/RelevantCancerTypesModal.tsx

```diff
@@ -0,0 +1,241 @@
+import { faUndo } from '@fortawesome/free-solid-svg-icons';
+import DefaultBadge from '../badge/DefaultBadge';
+import { CancerType } from '../model/firebase/firebase.model';
+import { RelevantCancerType } from './relevant-cancer-types-modal-store';
+import { CancerType as FetchedCancerType } from '../api/generated';
+import styles from './styles.module.scss';
+import { FaExclamationCircle } from 'react-icons/fa';
+
+export interface IRelevantCancerTypesModalProps extends StoreProps {
+  onConfirm: (newRelevantCancerTypes: CancerType[]) => void;
+  onCancel: () => void;
+}
+
+const RelevantCancerTypesModal = observer((props: IRelevantCancerTypesModalProps) => {
+  return props.relevantCancerTypesModalStore.isOpen ? <RelevantCancerTypesModalContent {...props} /> : <></>;
+});
+
+const RelevantCancerTypesModalContent = observer(
+  ({ relevantCancerTypesModalStore, onConfirm, onCancel }: IRelevantCancerTypesModalProps) => {
+    const allCancerTypesDeleted =
+      relevantCancerTypesModalStore.relevantCancerTypes.length > 0 &&
+      relevantCancerTypesModalStore.relevantCancerTypes.every(rct => rct.isDeleted);
+
+    useEffect(() => {
+      function convertFetchedCancerTypeToRelevantCancerType(cancerType: FetchedCancerType, isDeleted: boolean) {
+        const newRct = new RelevantCancerType();
+        newRct.mainType = cancerType.mainType;
+        newRct.subtype = cancerType.subtype;
+        newRct.code = cancerType.code;
+        newRct.isDeleted = isDeleted;
+        newRct.level = cancerType.level;
+        return newRct;
+      }
+
+      async function setRelevantCancerTypes() {
+        const fetchedRelevantCancerTypes = (
+          await cancerTypeClient.getRelevantCancerTypes(
+            relevantCancerTypesModalStore.tumor.cancerTypes.map(type => ({
+              mainType: type.mainType,
+              code: type.code,
+            })),
+            relevantCancerTypesModalStore.level ? `LEVEL_${relevantCancerTypesModalStore.level}` : null
+          )
+        ).data;
+
+        if (!relevantCancerTypesModalStore.firebaseRelevantCancerTypes) {
+          relevantCancerTypesModalStore.setRelevantCancerTypes(
+            fetchedRelevantCancerTypes.map(rct => convertFetchedCancerTypeToRelevantCancerType(rct, false))
+          );
+          return;
+        }
+
+        const rcts = fetchedRelevantCancerTypes.map(rct => {
+          for (const firebaseRct of relevantCancerTypesModalStore.firebaseRelevantCancerTypes) {
+            if ((rct.code && rct.code === firebaseRct.code) || (!rct.code && !firebaseRct.code && rct.mainType === firebaseRct.mainType)) {
+              return convertFetchedCancerTypeToRelevantCancerType(rct, false);
+            }
+          }
+          return convertFetchedCancerTypeToRelevantCancerType(rct, true);
+        });
+        relevantCancerTypesModalStore.setRelevantCancerTypes(rcts);
+      }
+
+      if (cancerTypeClient) {
+        setRelevantCancerTypes();
+      }
+    }, [cancerTypeClient]);
```

Comment:
This is a very interesting way. Would cancerTypeClient ever change?

Reply:
No it should not, I think I originally just put it there since it was a dependency I used, but it's a constant export, so I can't see any scenario where it changes.